contracts/whitelist/src/contract.rs

```diff
@@ -157,35 +178,32 @@ pub fn execute_update_end_time(
         .add_attribute("sender", info.sender))
 }
 
-pub fn execute_update_members(
+pub fn execute_add_members(
     deps: DepsMut,
     _env: Env,
     info: MessageInfo,
-    msg: UpdateMembersMsg,
+    msg: AddMembersMsg,
 ) -> Result<Response, ContractError> {
     let mut config = CONFIG.load(deps.storage)?;
     if info.sender != config.admin {
         return Err(ContractError::Unauthorized {});
     }
 
-    for add in msg.add.into_iter() {
+    for add in msg.to_add.into_iter() {
         if config.num_members >= MAX_MEMBERS {
             return Err(ContractError::MembersExceeded {
                 expected: MAX_MEMBERS,
                 actual: config.num_members,
             });
         }
         let addr = deps.api.addr_validate(&add)?;
+        if WHITELIST.has(deps.storage, addr.clone()) {
+            return Err(ContractError::DuplicateMember(addr.to_string()));
+        }
         WHITELIST.save(deps.storage, addr, &true)?;
         config.num_members += 1;
     }
 
-    for remove in msg.remove.into_iter() {
-        let addr = deps.api.addr_validate(&remove)?;
-        WHITELIST.remove(deps.storage, addr);
-        config.num_members -= 1;
-    }
-
```

Comment:
not totally sure if we should remove this, what if someone actually messed up and needs to delete a single account? They will need to recreate an entire whitelist just because a single mistake.  However maybe allow removing only if the whitelist time hasn't started.

Reply:
I felt that it over-complicates things. But you have a point.

Reply:
A separate message makes a lot more sense and is less error prone.